base/bin/build.sh

```diff
@@ -1,27 +1,188 @@
+#!/bin/bash
 
-# Change the current working directory to the `base` directory instead of
-# the `base/bin` directory
-pushd `dirname $0`/.. > /dev/null
+# fail-fast
+set -e
 
-RUNTIME_NAME="nodejs"
-DOCKER_NAMESPACE=${1}
-CANDIDATE_NAME=${2}
+function fail() {
+  echo "ERROR: ${1}"
+  exit 1
+}
+
+function print() {
+  if [ "${VERBOSE}" = "true" ]; then
+    echo "${1}"
+  fi
+}
+
+_NODE_VERSION_OPTION="--node-version"
+_UPLOAD_TO_STAGING_OPTION="--upload-to-staging"
+_CANDIDATE_NAME_OPTION="--candidate-name"
+_DOCKER_NAMESPACE_OPTION="--docker-namespace"
+_VERBOSE_OPTION="--verbose"
+_HELP_OPTION="--help"
+
+_DEFAULT="default"
```

Comment:
I've updated the code to only accept options of the form `--arg=value` (except for the `--verbose` and `--help` options), and I've updated the help text to reflect the changes.

Then if the script is invoked as `./base/bin/build.sh --docker-namespace=some-namespace`, then `some-namespace` will be used as the Docker namespace.

Otherwise if the script is invoked as `./base/bin/build.sh --docker-namespace=` or `./base/bin/build.sh --docker-namespace=""`, the script will act as if the Docker namespace was not specified and use the default value.

This allows the script to be easily used both directly or in other scripts.